From 045d73f5f39cfc9f8ebdcd90180ca807d46f61ca Mon Sep 17 00:00:00 2001
From: Matthias Proske <matthias.p@variscite.com>
Date: Mon, 27 Jul 2026 00:21:47 +0200
Subject: [PATCH] ti-apps-launcher: create profile for VAR-SOM-AM62P

This adds a dedicated profile for the VAR-SOM-AM62P.
This is based on the VAR-SOM-AM62 profile, but correctly matches the
dedicated Device Tree and shows the correct platform name.

Included apps are identical to VAR-SOM-AM62 as there are no obvious
additions from the AM62PX_EVK that should be added.

Signed-off-by: Matthias Proske <matthias.p@variscite.com>
---
 backend/includes/common.h |  1 +
 configs/am62px-var-som.h  | 36 ++++++++++++++++++++++++++++++++++++
 configs/platform.cpp      |  2 ++
 3 files changed, 39 insertions(+)
 create mode 100644 configs/am62px-var-som.h

diff --git a/backend/includes/common.h b/backend/includes/common.h
index e1c0877..7f5be68 100644
--- a/backend/includes/common.h
+++ b/backend/includes/common.h
@@ -29,6 +29,7 @@ struct device_info {
 };
 
 enum devices {
+    AM62PX_VAR_SOM,
     AM62PXX_EVM,
     AM62XX_EVM,
     AM62XX_LP_EVM,
diff --git a/configs/am62px-var-som.h b/configs/am62px-var-som.h
new file mode 100644
index 0000000..0c43661
--- /dev/null
+++ b/configs/am62px-var-som.h
@@ -0,0 +1,27 @@
+/* Configuration file for Variscite VAR-SOM-AM62P */
+
+#include "backend/includes/common.h"
+
+struct device_info device_info_am62px_var_som = {
+    .dtMatchString = "VAR-SOM-AM62P",
+    .platform = "am62px-var-som",
+    .wallpaper = "/images/am6x_oob_demo_home_image.png",
+    .include_apps = {
+        app_industrial_control_sitara,
+        app_live_camera,
+        app_arm_analytics,
+        app_benchmarks,
+        app_gpu_performance,
+        app_seva_store,
+        app_chromium_browser,
+        app_3d_demo,
+        app_settings,
+    },
+    .include_powerbuttons = {
+        action_shutdown,
+        action_reboot,
+    #if RT_BUILD == 0
+        action_suspend,
+    #endif
+    },
+};
diff --git a/configs/platform.cpp b/configs/platform.cpp
index 97f43fd..5982858 100644
--- a/configs/platform.cpp
+++ b/configs/platform.cpp
@@ -16,6 +16,7 @@
 
 #include "config_common.h"
 
+#include "am62px-var-som.h"
 #include "am62pxx-evm.h"
 #include "am62xx-evm.h"
 #include "am62xx-lp-evm.h"
@@ -52,6 +53,7 @@ RunCmd *chromium_browser = new RunCmd("/usr/bin/chromium-weston",
     chromium_missing_msg);
 
 QMap<enum devices, struct device_info> deviceMap = {
+    { AM62PX_VAR_SOM, device_info_am62px_var_som },
     { AM62PXX_EVM, device_info_am62p },
     { AM62XX_EVM, device_info_am62 },
     { AM62XX_LP_EVM, device_info_am62_lp },
-- 
2.55.0

